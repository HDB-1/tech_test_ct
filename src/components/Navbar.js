import React from 'react'
import {Link} from 'react-router-dom'
function NavBar() {
    return (
        <div>
            <Link to="/" >HOME</Link>
            <Link to="/info">Information</Link>
        </div>
    )
}

export default NavBar;