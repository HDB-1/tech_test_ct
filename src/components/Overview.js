import React from 'react'


export default function Overview(props) {
    return (
        <div>
            <p>showing overview component</p>
            <ul>
                {props.info.map((information, index) => <li key={index}>Name: {information.name}</li>)}
            </ul>
        </div>
    )
}

