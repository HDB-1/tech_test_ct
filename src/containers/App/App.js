import React from 'react';
import './App.css';
import { BrowserRouter as Router , Switch, Route } from 'react-router-dom';
import axios from 'axios';
import Overview from '../../components/Overview';
import Navbar from '../../components/Navbar';

class App extends React.Component {

  constructor(props){
    super(props);
    this.state = {
      data: [],
      fetched: false
    }
  }

  getAllData(){
    axios.get('https://demo.staging.energyhelpline-aws.com/api/results/a8823b4b-1abe-41de-a5b3-ab6700c08d98')
    .then((res) => {
      const all_data = res.data.elecResults;
      const shortened_res = all_data.slice(1, 20)
      this.setState({
        data: shortened_res,
        fetched: true
      })
      console.log(this.state.data)
    })
    .catch((err) => {
      console.log('this is the error: ' + err)
    })
  }

  componentDidMount(){
    this.getAllData();
  }

  render() {
    return (
      <Router className="App">
        <Navbar />
        <Switch>
          <Route exact path="/">
            <p>Welcome to the App</p>
          </Route>
          <Route exact path="/info">
          {this.state.fetched ? <Overview info={this.state.data}/> : 'Sorry there was a problem getting the data!'} 
          </Route>
        </Switch>
      </Router>
    );
  }
}

export default App;
