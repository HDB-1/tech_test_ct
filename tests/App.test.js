import React from 'react';
import { shallow } from 'enzyme';
import App from '../src/containers/App/App';
import Navbar from '../src/components/Navbar';
import Overview from '../src/components/Overview';
import waitUntil from "async-wait-until";

describe("shallow app", () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<App />)
    })
    it("should render correctly", () => {
        expect(wrapper).toMatchSnapshot();
    })
    it("Should render Navbar component", () => {
        expect(wrapper.containsMatchingElement(<Navbar />)).toEqual(true);
    })
    // it('should render an Overview component if the fetch was successful'),
    // async () => {
    //     wrapper.setState({fetched: true})
    //     await waitUntil(() => wrapper.instance().state.fetched)
    //     expect(wrapper.containsMatchingElement(<Overview />)).toEqual(true);
    // }
});

