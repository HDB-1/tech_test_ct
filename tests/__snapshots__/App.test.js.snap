// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow app should render correctly 1`] = `
<BrowserRouter
  className="App"
>
  <NavBar />
  <Switch>
    <Route
      exact={true}
      path="/"
    >
      <p>
        Welcome to the App
      </p>
    </Route>
    <Route
      exact={true}
      path="/info"
    >
      Sorry there was a problem getting the data!
    </Route>
  </Switch>
</BrowserRouter>
`;
